### PR TITLE
Use the web search api of the big register

### PR DESCRIPTION
**Pull request by davidv1992**
Changes to make the issuer check for birthday again using the web search.

### CODE REVIEW

File: src/main/java/foundation/privacybydesign/bigregister/RestApi.java [hunk 1]
```diff
@@ -85,10 +85,10 @@ public Response getAttributes(String disclosureJWT) throws KeyManagementExceptio
...
         // These attributes can be set for debugging purposes - to impersonate someone else.
         // Must only be used in a test environment!
-        //initials = "";
-        //familyName = "";
-        //gender = "";
-        //dateOfBirthString = "";
+        //String initials = "";
+        //String familyName = "";
+        //String gender = "";
+        //String dateOfBirthString = "";
```
Comment (aykevl) [inline src/main/java/foundation/privacybydesign/bigregister/RestApi.java:91]: This is unrelated to the PR and mostly a matter of taste I guess. The previous behavior was correct as long as the lines above (line 81-84) were not commented out.

File: src/main/java/foundation/privacybydesign/bigregister/RestApi.java [hunk 2]
```diff
@@ -98,9 +98,24 @@ public Response getAttributes(String disclosureJWT) throws KeyManagementExceptio
             return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity("Could not parse date").build();
         }
 
+        String bigNumber;
+        try {
+            bigNumber = new BIGWebSearch().getBIGNumber(familyName, dateOfBirth, gender);
+        } catch (BIGRequestException e) {
+            // This should indicate a problem on their end or with the connection, not on our side.
+            System.out.println("BIG request error: " + e.getMessage());
```
Comment (aykevl) [inline src/main/java/foundation/privacybydesign/bigregister/RestApi.java]: Nitpick: this should be `logger.error`, but as previously discussed this is probably originally my mistake.

**Review by aykevl — APPROVED** [review]
Looks good to me, can be merged in my opinion. It's even better than what I had originally in mind.

File: src/main/java/foundation/privacybydesign/bigregister/BIGWebSearch.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        StringBuilder queryBuilder = new StringBuilder();
+        // Name is always present
+        queryBuilder.append("name=");
+        try {
+            queryBuilder.append(URLEncoder.encode(name, "UTF-8"));
+        } catch (UnsupportedEncodingException e) {
+            throw new BIGRequestException("Unexpected encoding error: " + e.getMessage());
+        }
+        // Date if available
+        if (dateOfBirth != null) {
```
Comment (sietseringers) [inline src/main/java/foundation/privacybydesign/bigregister/BIGWebSearch.java]: Why would the birthdate not be available? If it isn't, then we shouldn't issue anything at all. That means that by this point in the code we can assume there is a birthdate.

File: src/main/java/foundation/privacybydesign/bigregister/BIGWebSearch.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+            queryBuilder.append("&dateOfBirth=");
+            queryBuilder.append(new SimpleDateFormat("dd-MM-yyyy").format(dateOfBirth));
+        }
+        // Gender if available
+        if (gender.equals("male")) {
+            queryBuilder.append("&gender=1");
+        } else if (gender.equals("female")) {
+            queryBuilder.append("&gender=2");
+        }
```
Comment (sietseringers) [inline src/main/java/foundation/privacybydesign/bigregister/BIGWebSearch.java]: I don't know if a bank will ever give anyone a gender different than these two, but if so we should probably have an `else` here that at least logs something, and maybe abort the request.

File: src/main/java/foundation/privacybydesign/bigregister/RestApi.java [hunk 5]
```diff
@@ -98,9 +98,24 @@ public Response getAttributes(String disclosureJWT) throws KeyManagementExceptio
             return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity("Could not parse date").build();
         }
 
+        String bigNumber;
+        try {
+            bigNumber = new BIGWebSearch().getBIGNumber(familyName, dateOfBirth, gender);
```
Comment (sietseringers) [inline src/main/java/foundation/privacybydesign/bigregister/RestApi.java]: `BIGWebSearch` has no instance members, so you can make `BIGWebSearch.getBIGNumber()` a static method. Then you don't have to create a `BIGWebSearch` here.

**Review by sietseringers — CHANGES_REQUESTED** [review]
Compilation currently fails because the `BIGRegisterApp` class calls `BIGService.doRequest()`  with too many parameters:
```
BIGRegisterApp.java:26: error: method doRequest in class BIGService cannot be applied to given types;
            results = service.doRequest(name, null, "");
```

**Comment by davidv1992** [comment]
Made requested changes

**Pull request merged**